apps/previewgenerator.sh

```diff
@@ -16,67 +16,209 @@ debug_mode
+                occ_command config:system:set enabledPreviewProviders 7 --value="OC\\Preview\\TXT"
+            ;;
+
+            Movie)
+                occ_command config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\Movie"
+            ;;
+
+            Photoshop)
+                occ_command config:system:set enabledPreviewProviders 15 --value="OC\\Preview\\Photoshop"
+            ;;
+
+            SVG)
+                occ_command config:system:set enabledPreviewProviders 18 --value="OC\\Preview\\SVG"
+            ;;
+
+            TIFF)
+                occ_command config:system:set enabledPreviewProviders 19 --value="OC\\Preview\\TIFF"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 9< results
+    rm -f results
+    clear      
+
+else
+    # Choose file formats fo the case when imagick is not installed.
+    whiptail --title "Choose file formats" --checklist --separate-output "Now you can choose for which file formats you would like to generate previews\nSelect or unselect by pressing the spacebar" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    "PNG" ON \
+    "JPEG" ON \
+    "GIF" ON \
+    "BMP" ON \
+    "MarkDown" ON \
+    "MP3" ON \
+    "TXT" ON \
+    "Movie" OFF 2>results
+
+    while read -r -u 9 choice
+    do
+        case $choice in
+            PNG)
+                clear
+                occ_command config:system:set enabledPreviewProviders 0 --value="OC\\Preview\\PNG"
+            ;;
+
+            JPEG)
+                occ_command config:system:set enabledPreviewProviders 1 --value="OC\\Preview\\JPEG"
+            ;;
+
+            GIF)
+                occ_command config:system:set enabledPreviewProviders 2 --value="OC\\Preview\\GIF"
+            ;;
+
+            BMP)
+                occ_command config:system:set enabledPreviewProviders 3 --value="OC\\Preview\\BMP"
+            ;;
+
+            MarkDown)
+                occ_command config:system:set enabledPreviewProviders 5 --value="OC\\Preview\\MarkDown"
+            ;;
+
+            MP3)
+                occ_command config:system:set enabledPreviewProviders 6 --value="OC\\Preview\\MP3"
+            ;;
+
+            TXT)
+                occ_command config:system:set enabledPreviewProviders 7 --value="OC\\Preview\\TXT"
+            ;;
+
+            Movie)
+                occ_command config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\Movie"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 9< results
+    rm -f results
+    clear
+fi
+
+# Enable additional previews (most likely not working; remove the # to enable the specific preview)
+#occ_command config:system:set preview_libreoffice_path --value="/usr/bin/libreoffice"
+#occ_command config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\XBitmap"
+#occ_command config:system:set enabledPreviewProviders 8 --value="OC\\Preview\\Illustrator"
+#occ_command config:system:set enabledPreviewProviders 10 --value="OC\\Preview\\MSOffice2003"
```

Comment:
Doesn't MSOffice and PDF work if you install the libreoffice packages (which you now seems to have removed)?

Reply:
No, I have tested this out and it hasn't worked also when libreoffice was installed
but look here for details:
https://github.com/nextcloud/vm/pull/1002#issuecomment-569828067

Reply:
Aah ok, thanks for confirming. 

Maybe remove those lines and refer to the Nextcloud documentation instead to keep it clean.

Reply:
I removed them and added the comment to the specific section (currently line 79 and 154)